src/server.ts

```diff
@@ -24,6 +24,25 @@ class ServerErrorGQL extends GraphQLError {
   }
 }
 
+async function getUser(userId: GetUserInput): Promise<User> {
+  let user: User;
+  try {
+    user = await prisma.user.findUnique({ where: { id: +userId.id } });
+  } catch (err) {
+    console.log(err);
+    throw new ServerErrorGQL(
+      400,
```

Comment:
aqui seria um not found - 404, e vc poderia encurtar o outro if se usar o findUniqueOrThrow

Reply:
Pelo que eu vi, o erro `404` serve para URLs ou endpoints que não são encontrados.
Como o erro é de uma entrada no banco de dados que não foi encontrado, e não um problema com o endpoint, não me parece fazer sentido.

Eu também escolhi fazer deste jeito (checando pelo valor `null` em vez de lançar um outro erro) para ter mensagens de erro diferentes, uma vez que buscar por um ID inválido é um caso comum e faz sentido deixar claro que o problema foi esse e não algum erro interno do back-end ou similar.